src/c-writer.cc

```diff
@@ -1452,6 +1501,15 @@ void CWriter::WriteMultivalueType(const TypeVector& types) {
   Write(CloseBrace(), ";", Newline(), "#endif  /* ", name, " */", Newline());
 }
 
+void CWriter::WriteMultivalueParamTypes() {
+  for (TypeEntry* type : module_->types) {
+    FuncType* func_type = cast<FuncType>(type);
+    if (func_type->GetNumParams() > 1) {
+      WriteMultivalueType(func_type->sig.param_types);
```

Comment:
It looks like "multivalue" is referring to any function type that takes more than one param?  Is that right?   If so I think it might be best to choose a different term since "multivalue" has a specific existing meaning in the Wasm world.

Reply:
Also. do we need to write out all these types, even for functions that are not tail callable?

Reply:
First comment was done in #2296. Just pushed a commit to this PR (269e5d83110b5e3bca43f780f13ccc4899dd62d8) for the second. The type structs are now only declared when necessary.